sdk/core/azure-core/src/http/curl/curl_connection_private.hpp

```diff
@@ -132,10 +132,12 @@ namespace Azure { namespace Core { namespace Http {
 #pragma warning(push)
 // C26812: The enum type 'CURLcode' is un-scoped. Prefer 'enum class' over 'enum' (Enum.3)
 #pragma warning(disable : 26812)
-#endif
         auto result = curl_easy_getinfo(m_handle, CURLINFO_ACTIVESOCKET, &m_curlSocket);
-#if defined(_MSC_VER)
 #pragma warning(pop)
+#else
+      long socket;
+      auto result = curl_easy_getinfo(m_handle, CURLINFO_LASTSOCKET, &socket);
+      m_curlSocket = socket;
```

Comment:
CURLINFO_LASTSOCKET won't work in all platforms as per libcurl documentation:

https://curl.se/libcurl/c/CURLINFO_ACTIVESOCKET.html

https://github.com/curl/curl/blob/3085ccfae996bb0fa606d2c7bc6783dc15d76a30/docs/libcurl/opts/CURLINFO_ACTIVESOCKET.3#L41

This change would break other systems where CURLINFO_LASTSOCKET won't work.

Is this change also because you might be using libcurl version < 7.45 ?

Reply:
Please note that this code is in the else clause of the _MSC_VER ifdef. This code is indeed to be able to compile with curl versions < 7.45. Please revise CURLINFO_LASTSOCKET documentation where it is made clear that this is only the case for Win64 platforms: https://curl.se/libcurl/c/CURLINFO_LASTSOCKET.html

Reply:
I don't feel strong about using a deprecated API =\  

I think this could bring us pain in the future. We would also need some strategy for testing and documenting this.

Reply:
@vhvb1989 @jorgen

Can we decide which function to call based on the value of macro `LIBCURL_VERSION_MINOR`?

We call `CURLINFO_LASTSOCKET ` if  the value is < 45, otherwise call `CURLINFO_ACTIVESOCKET`.

This way, we won't be using any deprecated API and we're compatible with old version of libcurl.

Reply:
Sure thing.

Reply:
I have updated the patch and started using the libcurl version macros.